PelicanPlatform/pelican
# Add authorization to Prometheus /metrics endpoint

Closes #291 

* Added Bearer authorization to Prometheus `/metrics` endpoint and issue token for the scraper of both director and server itself, with periodic reloading.
* Refactored authorization logic in `web_ui` to make it more flexible, clear, and reusable.
* Changed LoadDirectorPublicKey to return key, not pointer to the key

## Testing Instructions
* Spin up `director` and you should expect `/metrics` requests are all successful, and their frequency should be around 15s.
* Change `Monitoring.TokenExpiresIn` and `Monitoring.TokenRefreshInterval` to smaller value to ensure token refreshment is successful
* Run `director`, `registry` and `origin`. At `origin` end, confirm that its `/metrics` requests are also all successful.
* Go to origin's web APIs and check auth works as expected. i.e. `/api/v1.0/prometheus` as well as `/metrics` (You should be able to see it through web only when you logged in)
